### ISSUE TITLE
WebAppClassLoader loadClass can throw NullPointerException for missing class

### ISSUE THREAD

**Issue by wong-git** [eclipse/jetty.project#3307]
`loadClass` is invoking `resolveClass` which throws a `NullPointerException` when `findClass` returns null.

This is with Jetty 9.4.14.v2018111 and Java 11

Also, shouldn't invocation of `resolveClass` be dependent on the value of the passed `resolve` parameter?
`
    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException
`
...
`                    webapp_class = this.findClass(name);
                    resolveClass(webapp_class);
`

**Comment by janbartel** [comment]
@wong-git  do you have a repro test webapp? For your question about resolveClass call, is this a question arising from code inspection, or do you have a use case where it is causing a problem?

**Comment by wong-git** [comment]
@janbartel I don't have a repro test webapp. It was causing a problem in our upgrade to Java 11 where `resolveClass` apparently now throws a `NullPointerException` if the class parameter is null. (Prior to 11 it was a native implementation.)

This was appearing during the course of a GWT implementation, which (GWT) was using reflection to determine whether a class instance to be serialized had a custom field serializer class.

Ultimately, I tracked the root cause down to a private subclass of `WebAppClassLoader` provided by GWT, `com.google.gwt.dev.shell.jetty.JettyLauncher`, which for `findClass` is returning a null if it cannot find the resource. I reported this to them: https://github.com/gwtproject/gwt/issues/9651

Code inspection led me to proposing that `resolveClass` should probably not be invoked if the `resolve` parameter is false.
